### PR TITLE
NET-4: new transport

### PR DESCRIPTION
**Pull request by krestkrest**
**- What I did**
Implemented new network on transport layer to replace old DHT transport
**- How to verify it**
run tests

### CODE REVIEW

File: network/hostnetwork/transport.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+		select {
+		case msg := <-h.transport.Packets():
+			if msg == nil {
+				log.Error("HostNetwork receiving channel is closed")
+				break
+			}
+			if msg.Error != nil {
+				log.Warnf("Received error response: %s", msg.Error.Error())
+			}
+			h.processMessage(msg)
```
Comment (dmitry-panchenko) [inline network/hostnetwork/transport.go]: We should process packet in another goroutine.

File: network/hostnetwork/transport.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+		return errors.New("double listen initiated")
+	}
+	go h.listen()
+	return h.transport.Start()
+}
+
+func (h *hostTransport) listen() {
+	for {
+		select {
+		case msg := <-h.transport.Packets():
```
Comment (dmitry-panchenko) [inline network/hostnetwork/transport.go]: Why `msg`? Could we name it `packet`?

File: network/hostnetwork/transport.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+	}
+}
+
+func (f future) GetRequest() network.Request {
+	request := transport.Future(f).Request()
+	return (*packetWrapper)(request)
+}
+
+// Listen start listening to network requests, should be started in goroutine.
+func (h *hostTransport) Listen() error {
```
Comment (dmitry-panchenko) [inline network/hostnetwork/transport.go]: `.Listen()` and `.Disconnect()` methods are very similar to Component's `.Start()` and `.Stop()` methods.
Could we transform transport to Component?

File: network/transport/transport.go [hunk 4]
```diff
@@ -68,6 +68,7 @@ func NewTransport(cfg configuration.Transport, proxy relay.Proxy) (Transport, er
 	case "KCP":
 		return newKCPTransport(conn, proxy, publicAddress)
 	default:
+		closePlease(conn)
```
Comment (dmitry-panchenko) [inline network/transport/transport.go]: `closeVerbose`?

File: network/hostnetwork/transport.go (hunk 2 above)
Comment (krestkrest) [reply network/hostnetwork/transport.go]: I wanted to distinguish it from package `packet`

File: network/interfaces.go [hunk 5]
```diff
@@ -60,17 +61,19 @@ type HostNetwork interface {
 	PublicAddress() string
 
 	// SendRequest send request to a remote node.
-	SendRequest(Request) (Future, error)
+	SendRequest(request Request, receiver core.RecordRef) (Future, error)
```
[the remark this answers is not in the document]
Comment (krestkrest) [reply network/interfaces.go]: argument names were added to specify that the second argument is the receiver's node identifier

**Pull request merged**